### PR TITLE
Build tooling: Add linting for package.json files

### PR DESCRIPTION
**Pull request by gziolo**
## Description
This PR updates `@wordpress/scripts` to the next minor version which exposed `lint-pkg-json` command. It exposed the same checks for `package.json` files for all packages maintained by Lerna. This brings the setup of Gutenberg and packages closer to each other before they get merged together in the near future.

I also reordered `scripts` section to make everything sorted by script name. I also pulled in `npm-run-all` package to make it easier to run groups of scripts.

## How has this been tested?
* `npm run lint-pkg-json`
* `npm run lint`

To make sure that this new linter works you can update any `package.json` file inside packages folder and change the order of properties or update `author` field, or remove `publishConfig` field.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by youknowriad** [comment]
Removed the milestone as we don't really care when this lands.

(Looks like there's some failing tests)

File: package.json [hunk 1]
```diff
@@ -85,6 +85,7 @@
 		"eslint-plugin-react": "7.7.0",
 		"extract-text-webpack-plugin": "4.0.0-beta.0",
 		"node-sass": "git://github.com/sass/node-sass.git#v4.7.0",
+		"npm-run-all": "4.1.3",
```
Comment (aduth) [inline package.json]: What is `npm-run-all` providing that isn't already possible with `concurrently` (also a dependency)? Can we go with one or the other and not both?

File: package.json [hunk 2]
```diff
@@ -117,36 +118,59 @@
...
 		}
 	},
+	"npmPackageJsonLintConfig": {
+		"extends": "@wordpress/npm-package-json-lint-config",
+		"rules": {
+			"require-publishConfig": "error",
+			"valid-values-author": [
+				"error",
+				[
+					"WordPress"
```
Comment (aduth) [inline package.json]: Nit: Out of curiosity, why did we decide upon `"WordPress"` as `author` when `wordpress-develop/package.json` specifies it as `"The WordPress Contributors"`.

https://github.com/WordPress/wordpress-develop/blob/ecb3302/package.json#L13

(cc @ntwb)

File: package.json [hunk 3]
```diff
@@ -117,36 +118,59 @@
...
 		"dev": "npm run build:packages && concurrently \"cross-env webpack --watch\" \"npm run dev:packages\"",
 		"dev:packages": "node ./bin/packages/watch.js",
-		"test": "npm run lint && npm run test-unit",
-		"test-php": "npm run lint-php && npm run test-unit-php",
-		"ci": "concurrently \"npm run lint && npm run build\" \"npm run test-unit:coverage-ci\"",
 		"fixtures:clean": "rimraf \"core-blocks/test/fixtures/*.+(json|serialized.html)\"",
 		"fixtures:server-registered": "docker-compose run -w /var/www/html/wp-content/plugins/gutenberg --rm wordpress ./bin/get-server-blocks.php > core-blocks/test/server-registered.json",
 		"fixtures:generate": "npm run fixtures:server-registered && cross-env GENERATE_MISSING_FIXTURES=y npm run test-unit",
-		"fixtures:regenerate": "npm run fixtures:clean && npm run fixtures:generate",
+		"fixtures:regenerate": "npm-run-all fixtures:clean fixtures:generate",
```
Comment (aduth) [inline package.json]: Is `npm-run-all` default to parallel or sequential execution? Sequential is pretty important for this task to ensure that `clean` occurs before (and does not overlap with) newly generated files.

See also `run-s`: https://www.npmjs.com/package/npm-run-all#-usage

File: package.json [hunk 4]
```diff
@@ -117,36 +118,59 @@
...
-		"prebuild": "check-node-version --package",
+		"prebuild": "npm run check-engines",
 		"build:packages": "rimraf ./packages/*/build ./packages/*/build-module && node ./bin/packages/build.js",
 		"build": "npm run build:packages && cross-env NODE_ENV=production webpack",
-		"lint": "eslint .",
-		"lint:fix": "eslint . --fix",
-		"lint-php": "docker-compose run --rm composer run-script lint",
-		"predev": "check-node-version --package",
+		"check-engines": "check-node-version --package",
+		"ci": "concurrently \"npm run lint && npm run build\" \"npm run test-unit:coverage-ci\"",
```
Comment (aduth) [inline package.json]: Minor: `npm-run-all` could be used here now in place of `npm run lint && npm run build`. I think they could occur in parallel as well? **Edit:** Though, the fact we're using `concurrently` separate here makes me think we needed them to occur in sequence. Sounds familiar, hmm..

File: package.json (hunk 2 above)
Comment (ntwb) [reply package.json]: Also: `Copyright 2011-2018 by the contributors` via

Comment (gziolo) [reply package.json]: Updated 👍

File: package.json (hunk 4 above)
Comment (gziolo) [reply package.json]: I spend some time trying to rewrite it with `npm-run-all` but failed, so I removed this dependency at all in favor of `concurrently` :)

**Review by ntwb — APPROVED** [review]

**Pull request merged**